tests/Application/SaveMappings/SaveMappingsUseCaseTest.php

```diff
@@ -140,4 +142,26 @@ public function testShouldShowInvalidEntityId(): void {
 		$this->assertTrue( $this->presenter->showedInvalidEntityId );
 	}
 
+	public function testShouldShowPermissionDenied(): void {
+		$useCase = new SaveMappingsUseCase(
+			$this->presenter,
+			new PermissionDeniedMappingRepository(),
+			[ self::VALID_PREDICATE ],
+			new BasicEntityIdParser(),
+			new MappingListSerializer()
+		);
+
+		$useCase->saveMappings(
+			'Q1',
+			[
+				[ 'predicate' => self::VALID_PREDICATE, 'object' => self::VALID_OBJECT ]
+			]
+		);
+
+		$this->assertEquals(
```

Comment:
Is there a better way to test this exception? Or is this too specific and it's better to just check a boolean flag like the other presenter methods in the Spy class?

Reply:
Looks fine at first glance